OData/src/System.Web.OData/OData/Formatter/Serialization/ODataCollectionSerializer.cs

```diff
@@ -52,13 +53,17 @@ public ODataCollectionSerializer(ODataSerializerProvider serializerProvider)
         public sealed override ODataValue CreateODataValue(object graph, IEdmTypeReference expectedType,
             ODataSerializerContext writeContext)
         {
-            if (graph == null)
+            var config = writeContext.Request.GetConfiguration();
```

Comment:
Make the code goes through as original logical. For example, if the config is null, to serialize the null collection will throw exception.